projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/Boost_Cache.php

```diff
@@ -343,26 +368,26 @@ public function delete_on_post_transition( $new_status, $old_status, $post ) {
 	public function delete_on_post_trash( $post_id, $old_status ) {
 		if ( $this->is_published( $old_status ) ) {
 			$post = get_post( $post_id );
-			$this->delete_cache_for_post( $post );
-			$this->delete_cache_for_post_terms( $post );
-			$this->delete_cache_for_front_page();
-			$this->delete_cache_for_author( $post->post_author );
+			$this->invalidate_cache_for_post( $post, Filesystem_Utils::DELETE_ALL );
+			$this->invalidate_cache_for_post_terms( $post, Filesystem_Utils::DELETE_ALL );
+			$this->invalidate_cache_for_front_page( Filesystem_Utils::DELETE_ALL );
+			$this->invalidate_cache_for_author( $post->post_author, Filesystem_Utils::DELETE_ALL );
```

Comment:
When a post goes to trash. I think we should only invalidate the archive pages instead of deleting them?

Reply:
Unfortunately, if the post that was just deleted was the only post in that taxonomy, we will need to delete the cache files that were restored.

WPSC does a rebuild of the post itself that was deleted, but then when the output buffer callback runs it discovers it's a 404 page and [deletes the restored files](https://github.com/Automattic/jetpack/blob/ca60e4cd27557e2cc872cd28843bef9bedf9d5e6/projects/plugins/super-cache/wp-cache-phase2.php#L2155).
We could do something similar with Boost? Deleting a post is such a rare event, I think we should just delete the cache files in the first place.

Reply:
I modified the code so it still deletes the cache files for the post, but does a rebuild of the taxonomy files.
The taxonomy rebuild files will be deleted by the garbage collection eventually, and only used for serving to users if they're less than 10 seconds old, so they'll expire quickly.